src/Capability/Registry/ReferenceHandler.php

```diff
@@ -85,10 +85,28 @@ private function prepareArguments(\ReflectionFunctionAbstract $reflection, array
         $finalArgs = [];
 
         foreach ($reflection->getParameters() as $parameter) {
-            // TODO: Handle variadic parameters.
             $paramName = $parameter->getName();
             $paramPosition = $parameter->getPosition();
 
+            // Handle variadic parameters
+            if ($parameter->isVariadic()) {
+                // For variadic parameters, collect all remaining arguments
+                $variadicArgs = [];
+                foreach ($arguments as $key => $value) {
+                    if (is_numeric($key) && $key >= $paramPosition) {
+                        try {
+                            $variadicArgs[] = $this->castArgumentType($value, $parameter);
+                        } catch (InvalidArgumentException $e) {
+                            throw RegistryException::invalidParams($e->getMessage(), $e);
+                        } catch (\Throwable $e) {
+                            throw RegistryException::internalError("Error processing variadic parameter `{$paramName}`: {$e->getMessage()}", $e);
+                        }
+                    }
+                }
+                $finalArgs[$paramPosition] = $variadicArgs;
+                continue;
+            }
```

Comment:
```suggestion
            if ($parameter->isVariadic()) {
                return $arguments;
            }
```

This is sufficient for a use case where the handler has `...$arguments`:

```
                $definition['handler'] = fn (...$arguments) => ($this->operationHandler)($mcpName, $arguments);
```

Could you enlighten me with your use case where you gather the type of the variadic argument? What's the use case?